cobigen-cli/src/main/java/com/devonfw/cobigen/cli/commands/CobiGenCommand.java

```diff
@@ -1,20 +1,74 @@
 package com.devonfw.cobigen.cli.commands;
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.Properties;
+
+import org.apache.maven.model.Model;
+import org.apache.maven.model.io.xpp3.MavenXpp3Reader;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.shared.utils.xml.pull.XmlPullParserException;
+
 import com.devonfw.cobigen.cli.constants.MessagesConstants;
 
+import net.sf.mmm.util.io.api.IoMode;
+import net.sf.mmm.util.io.api.RuntimeIoException;
+import net.sf.mmm.util.xml.base.XmlInvalidException;
 import picocli.CommandLine.Command;
+import picocli.CommandLine.IVersionProvider;
+
 
 /**
  * This class defines the main CobiGen command
  */
+
+
 @Command(description = MessagesConstants.WELCOME_MESSAGE, name = "cobigen", aliases = { "cg" },
-    mixinStandardHelpOptions = true, version = "CobiGen CLI 1.0", subcommands = { GenerateCommand.class })
+    mixinStandardHelpOptions = true,  subcommands = { GenerateCommand.class },versionProvider = CobiGenCommand.PropertiesVersionProvider.class)
 public class CobiGenCommand implements Runnable {
 
-    @Override
-    public void run() {
+	@Override
+	public void run() {
+
+		// Nothing to do here, this is the master command
+	}
+
+	static class PropertiesVersionProvider implements IVersionProvider {
+		@Override
+		public String[] getVersion() throws Exception {
+
+			MavenXpp3Reader reader = new MavenXpp3Reader();
+			Model model;
+			if ((new File("pom.xml")).exists())
+				model = reader.read(new FileReader("pom.xml"));
+			else
+				model = reader.read(
+						new InputStreamReader(CobiGenCommand.class.getResourceAsStream("/src/main/resources/pom.xml")));
```

Comment:
Actually, we need to read the pom.xml that is present on the cobigen-cli folder. On the first execution of the CLI, we extract that pom.xml next to the `cobigen.jar` file. You should read that pom, because in the future, we will be able to change versions directly to that pom.

If you take a look [here](https://github.com/devonfw/tools-cobigen/blob/master/cobigen-cli/src/main/java/com/devonfw/cobigen/cli/utils/CobiGenUtils.java#L230), you will see how to get the real path of the pom.xml.